### PR TITLE
feat: port @predictions to GraphQL Transformer v2

### PR DESCRIPTION
**Pull request by cjihrig**
#### Description of changes
This commit ports the `@predictions` directive to v2 of the GraphQL Transformer.

#### Description of how you validated changes
Unit tests and manual testing.

#### Checklist
- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-graphql-transformer-core/src/graphql-api.ts [hunk 1]
```diff
@@ -297,6 +297,7 @@ export class GraphQLApi extends GraphqlApiBase implements GraphQLAPIProvider {
     layers?: ILayerVersion[],
     role?: IRole,
     environment?: { [key: string]: string },
+    timeout?: Duration,
```
Comment (SwaySway) [inline packages/amplify-graphql-transformer-core/src/graphql-api.ts:300]: Is this change related to the predictions directive?

File: packages/amplify-graphql-predictions-transformer/src/graphql-predictions-transformer.ts [hunk 2]
```diff
@@ -0,0 +1,700 @@
...
+
+const actionToRoleAction = new Map([
+  ['identifyText', 'rekognition:DetectText'],
+  ['identifyLabels', 'rekognition:DetectLabels'],
+  ['translateText', 'translate:TranslateText'],
+]);
+
+export type PredictionsConfig = {
+  bucketName: string;
+};
```
Comment (SwaySway) [inline packages/amplify-graphql-predictions-transformer/src/graphql-predictions-transformer.ts:81]: nit: Should we store these and the `X-Amz-Target`s in separate utils folder?

**Review by SwaySway — APPROVED** [review]
lgtm!

File: packages/amplify-graphql-transformer-core/src/graphql-api.ts (hunk 1 above)
Comment (cjihrig) [reply packages/amplify-graphql-transformer-core/src/graphql-api.ts:300]: Yes. It's used [here](https://github.com/aws-amplify/amplify-cli/pull/7387/files#diff-e71c291954828fb3ee3f43caa98a7b430958d9e7816d2a28b0f49b64ebeb5824R407).
